pubsub/kafka/kafka.go

```diff
@@ -0,0 +1,169 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package kafka
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/Shopify/sarama"
+	"github.com/dapr/components-contrib/pubsub"
+	log "github.com/sirupsen/logrus"
+)
+
+// Kafka allows reading/writing to a Kafka consumer group
+type Kafka struct {
+	producer      sarama.SyncProducer
+	consumerGroup string
+	brokers       []string
+}
+
+type kafkaMetadata struct {
+	Brokers       []string `json:"brokers"`
+	PublishTopic  string   `json:"publishTopic"`
+	ConsumerGroup string   `json:"consumerGroup"`
+}
+
+type consumer struct {
+	ready    chan bool
+	callback func(msg *pubsub.NewMessage) error
+}
+
+func (consumer *consumer) Setup(sarama.ConsumerGroupSession) error {
+	close(consumer.ready)
+	return nil
+}
+
+func (consumer *consumer) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
+	for message := range claim.Messages() {
+		if consumer.callback != nil {
+			err := consumer.callback(&pubsub.NewMessage{
+				Topic: claim.Topic(),
+				Data:  message.Value,
+			})
+			if err == nil {
+				session.MarkMessage(message, "")
+			}
+		}
+	}
+	return nil
+}
+
+func (consumer *consumer) Cleanup(sarama.ConsumerGroupSession) error {
+	return nil
+}
+
+// NewKafka returns a new kafka pubsub instance
+func NewKafka() pubsub.PubSub {
```

Comment:
Please add logger argument to pass structured logger. 
example:
https://github.com/dapr/components-contrib/blob/b210ea0e7fba1b331b802448be5d93dc65d03ac8/pubsub/redis/redis.go#L36-L38